lib/MPIABI/src/MPIABI.jl

```diff
@@ -0,0 +1,79 @@
+module MPIABI
+
+using Preferences
+
+function known_abis()
+    return (:MicrosoftMPI, :MPICH, :OpenMPI, :MPItrampoline)
+end
+
+# const abi = begin
+#     _abi = @load_preference("abi", nothing)
+#     if isnothing(_abi)
+#        _abi = find_abi()
+#        @set_preference!("abi" => _abi)
+#     end
+#     _abi
+# end
+
+const abi = @load_preference("abi", Sys.iswindows() ? :MicrosoftMPI : :MPICH)
+
+function __init__()
+    if Sys.iswindows()
+        if abi != :MicrosoftMPI
+            @error """
+                The MPI ABI is not compatible with the current platform.
+                Please set the MPI ABI to :MicrosoftMPI.
+            """
+        end
+    end
+end
+
+function set_abi(abi)
+    if abi ∉ known_abis()
+        error("""
+            The MPI ABI $abi is not supported.
+            Please set the MPI ABI to one of the following:
+            $(known_abis())
+        """)
+    end
+    @set_preferences!("abi" => abi)
```

Comment:
In #524, I suggested the following:
```
#  binary   = "" (default) | "system" | "MPICH_jll" | "OpenMPI_jll" | "MicrosoftMPI_jll"
#  path     = "" (default) | top-level directory location
#  library  = "" (default) | library name/path | list of library names/paths
#  abi      = "" (default) | "MPICH" | "OpenMPI" | "MicrosoftMPI" | "unknown"
#  mpiexec  = "" (default) | executable name/path
```

Reply:
`path`, `library`, `mpiexec` are dependent on which actual MPI installation is being loaded and the intention is for this to be used to choose which MPI installation to be loaded. 

The question is: Do we bifurcate along the `_jll` package used or along the ABI used. Thinking about the library versioning issue I think we need to make this mean with `_jll` is being used by MPI.jl (which is a shame) I would like it to be the ABI. 

Right now if the ABI is set to `OpenMPI` we will load `OpenMPI_jll` as part of `LAMMPS_jll`, so that would not be compatible with `binary==system && abi==openmpi` IIUC, except if we also set `libmpi_path` for `OpenMPI_jll`?

But then we could still run into a library version mismatch?

Reply:
How do you figure out the ABI without knowing the library etc?

Reply:
That's why this package doesn't do it. It is a communication channel for ABI aware `JLL` packages.

I was thinking MPI.jl would figure out the details and then set it here.

Reply:
There are some advantages to setting them all in the same package:
- it's easier to ensure that they stay in sync (e.g. that the preferences are set in the same location)
- it might help avoid an unnecessary round of precompilation

Reply:
Yeah I making this up here as we go along :) 
I don't think we have a good idea of what the best practices should be.

The other extreme is to just use MPI.jl And make all the JLL packages depend on it

Reply:
I like the idea of a subpackage, but if e.g. Yggdrasil binaries built against MPICH are only valid for MPICH_jll and not the system MPICH, it would make sense to have `binary` in the subpackage.